File: sales-savvy-frontend/src/services/productService.js
import axios from "axios";

const API = import.meta.env.VITE_API_URL || "http://localhost:8080/api/products";

export const getAllProducts = () =>
    axios.get(API);

export const getProductById = (productId) =>
    axios.get(`${API}/${productId}`);

export const searchProducts = (keyword) =>
    axios.get(`${API}/search?keyword=${keyword}`);

export const getProductsByCategory = (categoryId) =>
    axios.get(`${API}/category/${categoryId}`);

File: SalesSavvy-Backend/src/main/java/com/salessavvy/app/services/ProductService.java
package com.salessavvy.app.services;

import java.util.List;

import com.salessavvy.app.dto.response.ProductResponseDTO;

public interface ProductService {

    List<ProductResponseDTO> getAllProducts();

    ProductResponseDTO getProductById(Integer productId);
    
    List<ProductResponseDTO> searchProducts(String keyword);
    
    List<ProductResponseDTO> getProductsByCategory(Integer categoryId);

}